internal/generate/generate.go

```diff
@@ -177,38 +175,76 @@ func NewCA(secretName, secretNamespace string) *CA {
 }
 
 // loadKeyPair populates c.CACert/c.CAKey from c.CACertBytes/c.CAKeyBytes.
-func (c *CA) loadKeyPair() error {
-	caCert, err := helpers.ParseCertificatePEM(c.CACertBytes)
-	if err != nil {
-		return fmt.Errorf("failed to parse CA cert PEM: %w", err)
+func (c *CA) loadKeyPair(caCertBytes []byte) error {
+	c.CACerts = nil
+
+	// Parse all certificates (if multiple are present) to support the case in
+	// which [CA.CACertBytes] contains a chain of certificates. The first
+	// certificate is assumed to be the intermediate CA to be used to sign the
+	// newly generated certificates.
+	remaining := bytes.TrimSpace(caCertBytes)
+	for {
```

Comment:
`helpers.ParseCertificatesPEM` (Notice the Certificate**s**), seems to do pretty much exactly the same. Is there are reason we don't use it? :thinking: